src/commitlint/linter/_linter.py

```diff
@@ -47,16 +54,30 @@ def lint_commit_message(
         console.verbose("commit message ignored, skipping lint")
         return True, []
 
+    validator_instances: List[CommitValidator] = []
+
+    if not disable_max_header_length:
+        validator_instances.append(
+            HeaderLengthValidator(
+                commit_message=commit_message,
+                **{"max_header_length": max_header_length},  # type: ignore
```

Comment:
suggestion: This will work too.
```suggestion
                max_header_length=max_header_length,
```